### PR TITLE
Add missing metrics for security policies into Mesh Dashboard

### PR DESCRIPTION
**Pull request by carolynhu**
https://github.com/istio/istio/issues/26535

This PR is focusing on adding missing security policies (PeerAuthentication, RequestAuthentication, AuthorizationPolicy) into mesh dashboard.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Comment by carolynhu** [comment]
https://snapshot.raintank.io/dashboard/snapshot/McXuLDgujb6qzZYc0vDU7p0qj3qea7tq

File: manifests/addons/dashboards/istio-mesh-dashboard.json [hunk 1]
```diff
@@ -667,18 +667,35 @@
...
+          "legendFormat": "# of AuthorizationPolicy",
+          "refId": "C"
         }
       ],
       "thresholds": "",
       "timeFrom": null,
       "timeShift": null,
-      "title": "Authentication Mesh Policies",
-      "type": "singlestat",
+      "title": "Mesh Security Policies",
```
Comment (carolynhu) [inline manifests/addons/dashboards/istio-mesh-dashboard.json:697]: Here I renamed the chart name, maybe we could just calling it `Security Policies`?

**Comment by carolynhu** [comment]
This chart has 3 types of data to show, which causes it a little bit hard to tell which line is for which in the main Mesh dashboard. Any suggestions on how could we better render/arrange the position of it?

File: manifests/addons/dashboards/istio-mesh-dashboard.json (hunk 1 above)
Comment (douglas-reid) [reply manifests/addons/dashboards/istio-mesh-dashboard.json:697]: Can we just separate these out into three different charts to match the rest of the row?

**Comment by douglas-reid** [comment]
> This chart has 3 types of data to show, which causes it is a little bit hard to tell which line is for which in the main Mesh dashboard. Any suggestions on how could we better render/arrange the position/size of it?

My main suggestion is to start an additional row for the security policies. We can fill the gap on the existing row with a query for other network policies, or similar, if need be.

**Comment by carolynhu** [comment]
what about this: https://snapshot.raintank.io/dashboard/snapshot/G1jY59JxDxtWivIiCxxsF7vE5YHUSqPl

**Comment by carolynhu** [comment]
https://snapshot.raintank.io/dashboard/snapshot/fY7VcLwtkLbblU0onmhxkB7BxnY1BP0r?orgId=2

File: manifests/addons/dashboards/istio-mesh-dashboard.json [hunk 2]
```diff
@@ -430,7 +430,7 @@
       "thresholds": "",
       "timeFrom": null,
       "timeShift": null,
-      "title": "Virtual Services",
+      "title": "VirtualService",
```
Comment (douglas-reid) [inline manifests/addons/dashboards/istio-mesh-dashboard.json:433]: nit: i'd leave all of these both plural and with spaces for readability.

so, `Virtual Services`, `Destination Rules`, `Gateways`, `Workload Entries`, `Service Entries`. For the security bits, this gets more awkward, but I think appending `Policies` would be ok.

Comment (carolynhu) [reply manifests/addons/dashboards/istio-mesh-dashboard.json:433]: Ok, I see, let me update those titles.

**Comment by carolynhu** [comment]
update titles: https://snapshot.raintank.io/dashboard/snapshot/xK9BXo90ndg24GKBVYpwD2ssp1YHzbUI

**Review by douglas-reid — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]
Is this the best format for this? It seems odd to have most of the resources but not all, and to mix them with traffic dashboards. Maybe we should have a single panel that is `by (type)`? Not sure, this is clearly highly opinionated/stylistic.

**Comment by douglas-reid** [comment]
@howardjohn we probably need to think about what a mesh dashboard should show as a "first glance". having it all be different lines on a single chart has drawbacks as well.

@carolynhu you'll need to update the dashboard tests to account for these new queries (would be fine, i think to add them to the filter).

**Pull request merged**